### PR TITLE
Image sharer issue 3

### PR DESCRIPTION
**Pull request by chris-mason**
Closes #3

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,5 +3,7 @@ class ApplicationController < ActionController::Base
 
   def home
     @name = 'Marc Jacobs'
+
+    @images = Image.all.reverse
```
Comment (infiniteluke) [inline app/controllers/application_controller.rb:7]: Let's use `.order(...)` here

File: test/controllers/application_controlller_test.rb [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
-  test 'should say hello' do
+  def test_should_say_hello
     get root_path
     assert_select 'h1', text: 'Hello Marc Jacobs'
   end
+
+  def test_get_all_images
+    Image.create! url: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/030.png'
+    get root_path
+    assert_select 'img', src: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/030.png'
```
Comment (infiniteluke) [inline test/controllers/application_controlller_test.rb:12]: let's add enough images to assert on the sort order

File: app/controllers/images_controller.rb [hunk 3]
```diff
@@ -18,6 +18,10 @@ def show
     @image = Image.find(params[:id])
   end
 
+  def index
+    @images = Image.all.reverse
```
Comment (infiniteluke) [inline app/controllers/images_controller.rb:22]: same here

File: coverage/.last_run.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "result": {
+    "line": 36.21,
+    "branch": 25.0
+  }
+}
```
Comment (infiniteluke) [inline coverage/.last_run.json:6]: remove these coverage files and if they persist via rubymine, let's add them to ignore in a separate commit

File: app/models/image.rb [hunk 5]
```diff
@@ -1,8 +1,25 @@
...
 class Image < ApplicationRecord
-  validates :url, presence: true, format: {
-    with: URI::DEFAULT_PARSER.make_regexp,
-    message: 'URL does not match expected format'
-  }
+  # Old validator for image upload
+  # validates :url, presence: true, format: {
+  #   with: URI::DEFAULT_PARSER.make_regexp,
+  #   message: 'URL does not match expected format'
+  # }
```
Comment (infiniteluke) [inline app/models/image.rb:9]: you can remove this

**Comment by chris-mason** [comment]
[Issue 3: Image Index](https://trello.com/c/MJyaUbC3/446-issue-3-image-index)

**Comment by chris-mason** [comment]
Resolved issues above noted by Luke

**Review by infiniteluke — APPROVED** [review]

**Pull request merged**